src/request/sign-transaction/SignTransaction.js

```diff
@@ -11,29 +11,44 @@ class SignTransaction { // eslint-disable-line no-unused-vars
      * @returns {Promise<SignedTransactionResult>}
      * @protected
      */
-    async _signTx(txRequest, passphraseOrPin) {
-        if (txRequest.type !== TransactionType.BASIC) {
-            throw new Error('Not yet implemented');
-        }
+    async _doSignTransaction(txRequest, passphraseOrPin) {
+        if (txRequest.type === TransactionType.BASIC) {
+            // eslint-disable-next-line object-curly-newline
+            const { value, fee, recipient, signer, validityStartHeight } = txRequest;
+
+            const key = await KeyStore.instance.get(signer, passphraseOrPin);
+            const tx = key.createTransaction(recipient, value, fee, validityStartHeight);
 
-        // eslint-disable-next-line object-curly-newline
-        const { value, fee, recipient, signer, validityStartHeight } = txRequest;
+            const signatureProof = Nimiq.SignatureProof.unserialize(new Nimiq.SerialBuffer(tx.proof));
 
-        const key = await KeyStore.instance.get(signer, passphraseOrPin);
-        const tx = key.createTransaction(recipient, value, fee, validityStartHeight);
+            return {
+                type: TransactionType.BASIC,
+                sender: tx.sender.toUserFriendlyAddress(),
+                signerPubKey: signatureProof.publicKey.serialize(),
+                recipient: tx.recipient.toUserFriendlyAddress(),
+                value: Nimiq.Policy.satoshisToCoins(tx.value),
+                fee: Nimiq.Policy.satoshisToCoins(tx.fee),
+                validityStartHeight: tx.validityStartHeight,
+                signature: signatureProof.signature.serialize(),
+                hash: tx.hash().toBase64(),
+            };
+        }
+
+        if (txRequest.type === TransactionType.EXTENDED) {
```

Comment:
extended not implemented as it seems. suggest to throw an exception "not implemented"

Reply:
The check and thus exception throwing is at the start of the request in `SignTransactionApi`.